### PR TITLE
New date formatting options

### PR DESCRIPTION
**Pull request by emmajam**
Due to changes in the Japanese date standards the following new options are required 

New Format Example - 2019年4月24日

Add in the options to: 
- showYearFirst (show the year before the day or month is shown YYYY/MM/DD)
- spaceBeforeMonth (hide the space before the month value)
- spaceBeforeDay (hide the space before the day value)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,5 @@ npm-debug.log
 #Jetbrains IDE files
 .idea
 *.iml
+
+package-lock.json
```
Comment (icambron) [inline .gitignore]: This should be included

**Comment by icambron** [comment]
The code here looks great! A couple things before I can merge it:

 * Can you remove the `dist/` stuff from the PR?
 * Can you unignore the package-lock file?

**Comment by emmajam** [comment]
hey @icambron done ✅  sorry for slow response!

**Comment by icambron** [comment]
Thanks! Merged in ea95d283b80bb85ea58447e5dcd9a1f552be71e0, released in 1.3.1. I'm having trouble getting the website rebuilt, so the release isn't on there yet, but you can't get it under the releases tab here on Github, or pull it from npm.

**Pull request closed without merging**